ts/client/src/client.ts

```diff
@@ -3374,64 +3372,230 @@ export class MangoClient {
     return await this.sendAndConfirmTransactionForGroup(group, [ix]);
   }
 
-  /**
-   * Example:
-   * For a stop loss on SOL, assuming SOL/USDC pair
-   * priceLowerLimit - e.g.
-   *
-   * @param group
-   * @param account
-   * @param sellMintPk - would be SOL mint
-   * @param priceLowerLimit - priceLowerLimit would be greater than priceUpperLimit e.g. if SOL is at 25$, then priceLowerLimit could be 22$
-   * @param buyMintPk - would be USDC mint
-   * @param priceUpperLimit - priceLowerLimit would be greater than priceUpperLimit e.g. if SOL is at 25$, then priceUpperLimit could be 0$
-   * @param maxSell - max ui amount of tokens to sell, e.g. account.getTokenBalanceUi(solBank)
-   * @param pricePremiumFraction - premium in % the liquidator earns for executing this stop loss, this can be the slippage usually found for a particular size plus some buffer
-   * @param expiryTimestamp - is epoch in seconds at which this stop loss should expire, set null if you want it to never expire
-   * @returns
-   */
-  public async tokenConditionalSwapStopLoss(
+  public async tcsTakeProfitOnDeposit(
     group: Group,
     account: MangoAccount,
-    sellMintPk: PublicKey,
-    priceLowerLimit: number,
-    buyMintPk: PublicKey | null,
-    priceUpperLimit: number | null,
-    maxSell: number | null,
-    pricePremiumFraction: number | null,
+    sellBank: Bank,
+    buyBank: Bank,
+    thresholdPriceUi: number,
+    thresholdPriceInSellPerBuyToken: boolean,
+    maxBuyUi: number | null,
+    maxSellUi: number | null,
+    pricePremium: number | null,
+    allowCreatingDeposits: boolean | null,
+    allowCreatingBorrows: boolean | null,
```

Comment:
When you want to take profit on a deposit, you don't need to pass in maxBuy -- you want to buy as much as you can!

Similarly, why can you override allowCreatingBorrows? You're taking profit, you shouldn't accidentally create a borrow in the token!

(same arguments for the other functions)

Reply:
Fixed all of these, please check, I have only kept 

allowCreatingBorrows for tcsStopLossOnBorrow and tcsTakeProfitOnBorrow, in case you are using margin

Reply:
Looks good. Maybe rename allowCreatingBorrows to allowMargin?